### PR TITLE
Added Romberg's method

### PR DESCRIPTION
**Pull request by bernatguillen**
I added Romberg's method. It works (not sure if it's very fast). I am working on the fast version with fewer checks and assertions. 

The reason I only start checking for accuracy after 1/3 of the steps have been taken is to avoid situations where the function is 0 almost everywhere and therefore the first few steps all give 0. Other than that the code is almost copying and pasting the wikipedia pseudocode (except indexes starting at 1).

### CODE REVIEW

File: src/NumericalIntegration.jl [hunk 1]
```diff
@@ -71,6 +76,39 @@ function integrate(x::AbstractVector, y::AbstractVector, ::SimpsonEvenFast)
     @inbounds return (x[2] - x[1]) * retval
 end
 
+function integrate(x::AbstractVector, y::AbstractVector, m::RombergEven)
+    @assert length(x) == length(y) "x and y vectors must be of the same length!"
+    @assert ((length(x) - 1) & (length(x) - 2)) == 0 "Need length of vector to be 2^n + 1"
+    println(size(y[1]))
+    maxsteps::Integer = Int(log2(length(x)-1))
+    rombaux = zeros(typeof(x[1]), maxsteps, 2)
```
Comment (dextorious) [inline src/NumericalIntegration.jl]: This line is the reason the StaticArrays tests fail, you can't assign a SVector to a Float64. There might be edge cases I'm not considering, but my first impulse would be to write `rombaux = zeros(eltype(y), maxsteps, 2)` there.

File: src/NumericalIntegration.jl [hunk 2]
```diff
@@ -71,6 +76,39 @@ function integrate(x::AbstractVector, y::AbstractVector, ::SimpsonEvenFast)
...
+        for j in 1 : npoints
+            c += y[1 + (2*j-1)*jumpsize]
+        end
+        rombaux[1, currrow] = h*c + HALF*rombaux[1, prevrow]
+        for j in 2 : (i+1)
+            n_k = 4^(j-1)
+            rombaux[j, currrow] = (n_k*rombaux[j-1, currrow] - rombaux[j-1, prevrow])/(n_k - 1)
+        end
+
+        if i > maxsteps//3 && abs(rombaux[i, prevrow] - rombaux[i+1, currrow]) < m.acc
```
Comment (dextorious) [inline src/NumericalIntegration.jl]: This will also fail for StaticArrays due to the `abs` call. Not sure on the exact error metric you want to use here (L2 or Linf?), but replacing `abs` with `norm` will make the tests pass. You can replace it with an infinity norm if you like, the point is it has to be able to deal with SVectors too.

**Review by dextorious — CHANGES_REQUESTED** [review]
Sorry for the late review. I noted the two reasons the StaticArrays tests fail right now. Fix those when you get a chance and let me know if you intend to add anything further to the PR before you want me to merge it.

File: src/NumericalIntegration.jl (hunk 1 above)
Comment (bernatguillen) [reply src/NumericalIntegration.jl]: Made the changes you suggested and now tests pass! Thank you very much :1st_place_medal:

**Comment by bernatguillen** [comment]
I don't plan to add a fast version, for now, I'm not sure how well fastmath would work with some of the operations (TBH not very familiar with fastmath).

**Comment by bernatguillen** [comment]
Last thing, added a warning at the end in case the extrapolation estimated accuracy isn't below the desired accuracy. It doesn't throw an exception but it is useful for logging. I also added a test to make sure the warning is properly raised.

**Comment by dextorious** [comment]
All right, looks good to me.
